### PR TITLE
Move splinebound calculation to Numerics.

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Intend to use this function in other spline classes.

## What type(s) of changes does this code introduce?
- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted

### CODE REVIEW

**Review by prckent — CHANGES_REQUESTED** [review]
Sensible to move to Numerics for wider use. I don't have a better more generic name for the function, although we should try to think of one.

1. Can you add some consts to this functions arguments? The argument ordering is non-standard with inputs first and last. If it is going to be used more widely making its use clearer will help.
2. Is the workaround for the Intel OneAPI compiler bug still needed?

**Comment by ye-luo** [comment]
1. changed the function pattern to make output last.
2. No idea for the moment. Not affecting host but used to impact when used in offload regions.

**Review by prckent — APPROVED** [review]

**Comment by prckent** [comment]
Test this please

File: src/spline2/MultiBsplineEval_helper.hpp [hunk 1]
```diff
@@ -20,74 +20,37 @@
...
 /** define computeLocationAndFractional: common to any implementation
  * compute the location of the spline grid point and residual coordinates
  * also it precomputes auxiliary array a, b and c
  */
 template<typename T>
-inline void computeLocationAndFractional(const typename qmcplusplus::bspline_traits<T, 3>::SplineType* restrict spline_m,
-                            T x, T y, T z,
-                            int& ix, int& iy, int& iz,
-                            T a[4], T b[4], T c[4])
+inline void computeLocationAndFractional(
```
Comment (PDoakORNL) [inline src/spline2/MultiBsplineEval_helper.hpp:33]: is this what clang-format produced?  If this is hand formatting please protect it with format on and off tags.

Comment (ye-luo) [reply src/spline2/MultiBsplineEval_helper.hpp:33]: Done by clang-format

**Pull request merged**